### PR TITLE
Meta: use Makefile from deploy.sh

### PR DESCRIPTION
**Pull request by annevk**
Avoids duplication and hopefully keeps strictness consistent.

@tobie @domenic @Ms2ger any reason these are not in sync? I noticed the build script not failing with https://github.com/heycam/webidl/pull/848 while it was failing locally.

### CODE REVIEW

**Comment by annevk** [comment]
Hmm, perhaps `node ./check-grammar.js` should move into the `Makefile` as well as I guess it happens at the wrong time now?

File: Makefile [hunk 1]
```diff
@@ -28,6 +29,11 @@ else
 		exit 22 \
 	);
 endif
+ifdef node_installed
+	node ./check-grammar.js index.html
+else
+	@echo You need node for grammer checking.
```
Comment (Ms2ger) [inline Makefile]: One comment, though... Do we want this to not happen silently in automation if something's off with the setup, or should it fail loudly?

**Comment by annevk** [comment]
It seems to work well now, but I guess a second check from @domenic wouldn't hurt.

File: Makefile (hunk 1 above)
Comment (annevk) [reply Makefile]: That's a good point, not sure. Perhaps it's easier to continue to duplicate...

Comment (domenic) [reply Makefile]: You could check if one of https://docs.travis-ci.com/user/environment-variables/#default-environment-variables is set. Figuring out how to do that would require figuring out how Makefile syntax works though -_-.

Comment (domenic) [reply Makefile]: Similarly that could help avoid CI echoing the whole "Bikeshed is not locally installed" thing above by wrapping that in an if statement.

Comment (annevk) [reply Makefile]: @sideshowbarker I suspect this is up your alley?

Comment (sideshowbarker) [reply Makefile]: > You could check if one of https://docs.travis-ci.com/user/environment-variables/#default-environment-variables is set. Figuring out how to do that would require figuring out how Makefile syntax works though -_-.
> @sideshowbarker I suspect this is up your alley?

Yeah, I’ll take a look

Comment (sideshowbarker) [reply Makefile]: > You could check if one of https://docs.travis-ci.com/user/environment-variables/#default-environment-variables is set. Figuring out how to do that would require figuring out how Makefile syntax works though -_-.

So I see that makefile is already using some `ifdef …`. The makefile syntax for checking environment variables is to just use, e.g. `ifdef $(TRAVIS)` — that’d seem to be the simplest way to identify whether we’re being run in the Travis environment. There’s also `ifndef` for the negation of that, as well as `ifeq` and `ifneq` for any case where we’d want to actually check the value of an environment variable (rather that just checking if it’s defined.)

(I’d push a change for this to the PR branch myself, but I don’t have push perms here.)

**Comment by sideshowbarker** [comment]
```diff
commit 4c9f51e332cb6522f98c34ed1f5f00b2eb62034d
Author: Michael[tm] Smith <mike@w3.org>
Date:   Fri Mar 13 16:20:43 2020 +0900

    Add some 'ifndef $(TRAVIS)'

diff --git a/Makefile b/Makefile
index e310d6a..77542fe 100644
--- a/Makefile
+++ b/Makefile
@@ -11,6 +11,7 @@ index.html : index.bs
 ifdef bs_installed
 	bikeshed spec --die-on=warning index.bs
 else
+ifndef $(TRAVIS)
 	@echo Can\'t find a local version of Bikeshed. To install it, visit:
 	@echo
 	@echo https://github.com/tabatkins/bikeshed/blob/master/docs/install.md
@@ -29,11 +30,14 @@ else
 		exit 22 \
 	);
 endif
+endif
 ifdef node_installed
 	node ./check-grammar.js index.html
 else
+ifndef $(TRAVIS)
 	@echo You need node for grammer checking.
 endif
+endif
 ifdef pp_webidl_installed
 	npm run pp-webidl -- --input index.html
 else ifdef npm_installed
```

**Pull request merged**